warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/NonShardedSplitsFile.java

```diff
@@ -153,49 +165,73 @@ private static boolean matchesFileName(String cutFileName, Path cacheFile) {
     
     public static class Reader {
         private Map<String,Text[]> splits;
+        private Map<String,Map<Text,String>> splitsAndLocations;
         
-        public Reader(Configuration conf, Path[] filesToCheck, boolean isTrimmed) throws IOException {
-            Path cacheFile = findSplitsFile(conf, filesToCheck, isTrimmed);
+        public Reader(Configuration conf, Path[] filesToCheck, SplitsFileType splitsFileType) throws IOException {
+            Path cacheFile = findSplitsFile(conf, filesToCheck, splitsFileType);
             if (null == cacheFile) {
                 throw new RuntimeException("Could not find cut point file");
             }
             
             splits = new HashMap<>();
+            splitsAndLocations = new HashMap<>();
             ArrayList<Text> cutPoints = new ArrayList<>();
+            SortedMap<Text,String> cutPointsAndLocations = new TreeMap<>();
             String previousTableName = null;
+            boolean isFirstLine = true;
+            boolean isSameTable;
+            boolean hasSplits = false;
+            boolean hasLocations = false;
+            
             try (BufferedReader in = new BufferedReader(new FileReader(cacheFile.toString()))) {
                 String line;
                 while ((line = in.readLine()) != null) {
                     String[] parts = line.split("\\t");
```

Comment:
since we are splitting on a single character, use StringUtils.split which is much more efficient